lib/src/executables/intl_message_migration.dart

```diff
@@ -187,61 +159,123 @@ void main(List<String> args) async {
   print('^ Ignore the "codemod found no files" warning above for now.');
 
   for (String package in packageRoots) {
-    _log.info('Starting migration for $package');
-
-    final packageRoot = p.basename(package);
-    final packageName = packageNameLookup[packageRoot] ?? 'fix_me_bad_name';
-    _log.info('Starting migration for $packageName');
-    final packageDartPath;
-    try {
-      packageDartPath =
-          dartFilesToMigrateForPackage(package, processedPackages).toList();
-    } on FileSystemException {
-      _log.info('${package} does not have a lib directory, moving on...');
-      continue;
-    }
-    sortPartsLast(packageDartPath);
-
-    final File outputFile = fs.currentDirectory.childFile(
-        p.join(package, 'lib', 'src', 'intl', '${packageName}_intl.dart'));
-    final bool existingOutputFile = outputFile.existsSync();
-
-    final className = toClassName('${packageName}');
-    final classPredicate =
-        "import 'package:intl/intl.dart';\n\n//ignore: avoid_classes_with_only_static_members\nclass $className {\n";
-    if (!existingOutputFile) {
-      outputFile.createSync(recursive: true);
-      outputFile.writeAsStringSync(classPredicate);
-    } else {
-      final List<String> lines = outputFile.readAsLinesSync();
-      lines.removeLast();
-      String outputContent = lines.join('\n');
-      outputFile.writeAsStringSync(outputContent);
-    }
-
-    final intlPropMigrator = IntlMigrator(className, outputFile);
-    final displayNameMigrator = ConfigsMigrator(className, outputFile);
-    final importMigrator =
-        (FileContext context) => intlImporter(context, packageName, className);
-
-    exitCode = await runCodemodSequences(packageDartPath, [
-      [intlPropMigrator],
-      [displayNameMigrator],
-      [importMigrator]
-    ]);
-
-    processedPackages.add(package);
-    if (exitCode != 0 || outputFile.readAsStringSync() == classPredicate) {
-      outputFile.deleteSync();
-    } else {
-      List<String> lines = outputFile.readAsLinesSync();
-      final functions = lines.sublist(4);
-      functions.removeWhere((string) => string == '');
-      functions.sort();
-      lines.replaceRange(4, lines.length, functions);
-      lines.add('}');
-      outputFile.writeAsStringSync(lines.join('\n'), mode: FileMode.write);
-    }
+    await migratePackage(
+        package, packageNameLookup, processedPackages, codemodArgs, dartPaths);
+  }
+}
+
+void printUsage() {
+  stderr.writeln(
+      'Migrates literal strings that seem user-visible in the package by wrapping them in Intl.message calls.');
+  stderr.writeln();
+  stderr.writeln('Usage:');
+  stderr.writeln('    intl_message_migration [arguments]');
+  stderr.writeln();
+  stderr.writeln('Options:');
+  stderr.writeln(parser.usage);
+}
+
+/// Runs a set of codemod sequences separately to work around an issue where
```

Comment:
Moved from being defined as an inner function, and added codemodArgs to the parameters.